ddtrace/settings/config.py

```diff
@@ -50,12 +50,23 @@ def __init__(self):
 
         self.analytics_enabled = asbool(get_env("trace", "analytics_enabled", default=legacy_config_value))
 
+        self.tags = parse_tags_str(os.getenv("DD_TAGS") or "")
+
+        self.env = os.getenv("DD_ENV") or self.tags.get("env")
         # DEV: we don't use `self._get_service()` here because {DD,DATADOG}_SERVICE and
         # {DD,DATADOG}_SERVICE_NAME (deprecated) are distinct functionalities.
-        self.service = get_env("service")
+        self.service = os.getenv("DD_SERVICE") or os.getenv("DATADOG_SERVICE") or self.tags.get("service")
+        self.version = os.getenv("DD_VERSION") or self.tags.get("version")
```

Comment:
```suggestion
        self.version = os.getenv("DD_VERSION") or self.tags.pop("version", None)
```

Reply:
Would save you the check/removal below on L67

Reply:
The issue there is that if `DD_VERSION` is defined then the `pop` call will never happen

Reply:
oh duh, you are right, good catch